lib/kubeclient/common.rb

```diff
@@ -296,11 +296,15 @@ def get_entity(klass, resource_name, name, namespace = nil)
       new_entity(result, klass)
     end
 
-    def delete_entity(resource_name, name, namespace = nil)
+    def delete_entity(resource_name, name, namespace = nil, options = nil)
       ns_prefix = build_namespace_prefix(namespace)
+
+      params = {}
+      params.merge!(options[:query]) if options.include?(:query)
```

Comment:
yes. make sense. we can create a list of allowed queries like the get_entitites and use the same format.

it would be: 
gracePeriodSeconds = `grace_period_seconds`
orphanDependents = `orphan_dependents`
and
propagationPolicy = `propagation_policy`

if agree, I can change the code.